code/io/timer.h

```diff
@@ -63,6 +63,7 @@ class TIMESTAMP : public util::ID<timestamp_tag, int, -1>
 
 	inline bool isNever() const { return m_val == 0; }
 	inline bool isImmediate() const { return m_val == 1; }
+	inline bool isFinite() const { return isValid() && !isNever(); }
 };
```

Comment:
Honestly, these timestamp changes should be separate.

Reply:
Ok

Reply:
I've split them off into PR #6278